Modules/Challenges/test/SpecFlowChallenges.Specs/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -0,0 +1,219 @@
+using SpecFlowChallenges.Specs.Models;
+using TechTalk.SpecFlow.Assist;
+using static SpecFlowChallenges.Specs.Configuration.Settings;
+
+namespace SpecFlowChallenges.Specs.StepDefinitions;
+
+[Binding]
+public class ChallengesApiStepDefinitions
+{
+    private readonly IConfiguration _config;
+    private readonly RestClient _client;
+    private readonly ChallengesApi _challengeApi;
+    private string? _challengeId;
+    private bool _isAuthenticatedChallenge;
+    private RestResponse<ChallengeResponse> _challengeResponse;
+    private readonly RequestConfiguration _requestConfiguration;
+    private readonly AuthenticationParameters _authenticationParams;
+
+    public ChallengesApiStepDefinitions(IConfiguration config)
+    {
+        _config = config;
+        var settings = _config.GetSection(nameof(HttpConfiguration)).Get<HttpConfiguration>() ?? new HttpConfiguration();
+
+        _client = new RestClient(settings.BaseUrl);
+        _challengeApi = new ChallengesApi(_client);
+        _challengeId = string.Empty;
+        _challengeResponse = new RestResponse<ChallengeResponse>();
+        _requestConfiguration = new RequestConfiguration();
+        _authenticationParams = new AuthenticationParameters
+        {
+            Parameters = new Dictionary<string, string>
+            {
+                { "grant_type", "password" },
+                { "client_id", settings.ClientCredentials.ClientId },
+                { "client_secret", settings.ClientCredentials.ClientSecret },
+                { "username", "USRa" },
+                { "password", "a" }
+            }
+        };
+    }
+
+    [Given(@"the user is authenticated")]
+    public async Task GivenTheUserIsAuthenticated()
+    {
+        var accessTokenResponse = await _challengeApi.GetAccessTokenAsync(_authenticationParams);
+        _requestConfiguration.TokenResponse = accessTokenResponse.Data;
+
+        accessTokenResponse.IsSuccessStatusCode.Should().BeTrue();
+    }
+
+    [Given(@"the user is unauthenticated")]
+    public void GivenTheUserIsUnauthenticated()
+    {
+        _requestConfiguration.TokenResponse = null;
+    }
+
+    [Given(@"the Accept header is '([^']*)'")]
+    public void GivenTheAcceptHeaderIs(string acceptHeader)
+    {
+        _requestConfiguration.AcceptHeader = acceptHeader;
+    }
+
+    [Given(@"a Challenge c")]
+    public async Task GivenAChallengeC()
+    {
+        using (new AssertionScope())
+        {
+            var challengeResponse = await _challengeApi.CreateChallengeAsync(new RequestConfiguration());
+            challengeResponse.IsSuccessStatusCode.Should().BeTrue();
+
+            if (challengeResponse.Data != null)
+            {
+                _challengeId = challengeResponse.Data?.Result?.Id;
+                _challengeId.Should().NotBeNullOrEmpty(because: "Required value for 'Id' is missing.");
+            }
+            else
+            {
+                challengeResponse.Data.Should().NotBeNull();
```

Comment:
This assertion will **always** fail. So you could as well write it before the `if` and remove the `if` afterwards. But as stated above, you don't need the `if` at all, because at this place we can assume that the POST-route does its job.